XenAdmin/Controls/SrPickerItem.cs

```diff
@@ -231,12 +231,13 @@ public abstract class SrPickerItem : CustomTreeNode, IComparable<SrPickerItem>
         protected long DiskSize { get; private set; }
         protected readonly VDI[] existingVDIs;
 
-        protected SrPickerItem(SR sr, Host aff, long diskSize, VDI[] vdis)
+        protected SrPickerItem(SR sr, Host aff, VDI[] vdis)
         {
             existingVDIs = vdis ?? new VDI[0];
             TheSR = sr;
             Affinity = aff;
-            DiskSize = diskSize;
+            DiskSize = existingVDIs.Sum(vdi =>

```

Comment:
One thing I'm wondering about is whether this calculation should move closer to where SR.VdiCreationCanProceed is called.